### ISSUE TITLE
Can't change GatewayX509CertificateFindValue on existing Azure SF Cluster

### ISSUE THREAD

**Issue by Mardoxx** [Azure/service-fabric-issues#166]
### Reproduction Steps:
1. Setup cluster with reverse proxy (e.g. [default 5 VM 1 node type template](https://github.com/ChackDan/Service-Fabric/tree/master/ARM%20Templates)) with the additions made [here](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reverseproxy).
2. Deploy cluster
3. See that `https://<Cluster FQDN | internal IP>:19080` serves up correct certificate
4. Upload new cert to key vault
5. Install new certificates to VMs via VMSS ARM template. Edit appropriate `Microsoft.Compute/virtualMachineScaleSets` template and append new certificate values to `properties.virtualMachineProfile.osProfile.secrets.vaultCertificates`
6. Set new thumbprint in CLuster ARM template. Edit appropriate `Microsoft.ServiceFabric/clusters` template and set `properties.httpApplicationGatewayCertificate.thumbprint` to new value
7. Service fabric will fail with, e.g.

```Node name: _Default_0 has failed to open with upgrade domain: 0, fault domain: fd:/0, address: 10.0.0.4, hostname: Default000000, isSeedNode: true, versionInstance: 5.4.164.9494:1, id: 7ec9f3db8d79f7c701957617648293a0, dca instance: 131317337251927725, error: FABRIC_E_INVALID_CONFIGURATION```

SF Node `_Default_0` will be permanently down, with no obvious way to bring it back up. Even with restarting the services or rebooting the VM.

-----

### What happened:
I have my cluster up and running ([default 5 VM 1 node type template](https://github.com/ChackDan/Service-Fabric/tree/master/ARM%20Templates)) with the additions made [here](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reverseproxy). I wanted to change the SSL certificate used to protect my reverse proxy.

### Why I need this:
Reverse proxy will be secured with SSL cert, will inevitably need to update cert reference before it expires.

**Comment by Mardoxx** [comment]
I also tried the following:

1. Take down node (`_Default_0`) in SF Explorer
2. RDP into node
3. Stop MS Service Fabric Host service
4. Go in to `D:\SvcFab\_Default_0\Fabric\Fabric.Config.{latestDigest}`
5. Edit `GatewayX509CertificateFindValue` in Settings.xml
6. Attempt to start MS Service Fabric Host service

Same error occurs here.

If I set the `GatewayX509CertificateFindValue` back to its original value, it loads up fine.

I have also tried restarting the VM before step 6, this also does not work.

I do not understand the inner workings of SF well enough to give any further indication as to what is wrong. Are there any other diagnostic logs I can check?
